### PR TITLE
Rearrange tests in `html-metadata-test.js`

### PR DESCRIPTION
**Pull request by esanzgar**
I have created a few additional `describe` sub-sections and re-grouped
tests by the resulting metadata property. This allow for a more atomic
testing.

### CODE REVIEW

File: src/annotator/integrations/test/html-metadata-test.js [hunk 1]
```diff
@@ -177,83 +70,218 @@ describe('HTMLMetadata', () => {
...
       });
-    });
 
-    it('should ignore favicons with invalid URIs', () => {
-      tempDocumentHead.innerHTML = `
-        <link rel="favicon" href="http://a:b:c">
-      `;
-      const metadata = testDocument.getDocumentMetadata();
-      assert.isUndefined(metadata.favicon);
+      it('should ignore `pdf_url` links with invalid URIs', () => {
```
Comment (robertknight) [inline src/annotator/integrations/test/html-metadata-test.js:172]: This test is about the `metadata.link` property, but is not grouped with the other tests related to link parsing.

**Review by robertknight — COMMENTED** [review]
Since both `<link>` and `<meta>` elements can contribute to the `link` property of the result, there is ambiguity in the current grouping over which group some tests belong with. One way to avoid this would be to make both groups of tests about the output properties being checked, rather than the input element type that contribute to them.

File: src/annotator/integrations/test/html-metadata-test.js [hunk 2]
```diff
@@ -177,83 +70,218 @@ describe('HTMLMetadata', () => {
...
+          <link rel="shortlink" href="http://example.com/bookmark/short"></link>
+          <link rel="alternate" href="es/foo.html" hreflang="es" type="text/html"></link>
+          <link rel="icon" href="http://example.com/images/icon.ico"></link>
+          <link rel="canonical" href="http://example.com/canonical"></link>
+
+          <meta name="citation_doi" content="10.1175/JCLI-D-11-00015.1">
+          <meta name="citation_pdf_url" content="foo.pdf">
+          <meta name="dc.identifier" content="doi:10.1175/JCLI-D-11-00015.1">
+          <meta name="dc:identifier" content="foobar-abcxyz">
+          <meta name="dc.relation.ispartof" content="isbn:123456789">
```
[the remark this answers is not in the document]
Comment (esanzgar) [reply src/annotator/integrations/test/html-metadata-test.js:204]: Those five `<meta>` elements contribute to the `metadata.link` property.

**Review by esanzgar — COMMENTED** [review]
As suggested, I have re-grouped the test based on the output of the property being tested.

There is a lot of changes, but in reality it is a copy/paste exercise. In general, this grouping make it simpler to check the input and the output.

**Review by robertknight — APPROVED** [review]
I revised this slightly to avoid deeply nested `describe` blocks that have only a single test, and also clarify that the "link" tests pertain to properties other than `link` (`favicon`, `documentFingerprint`).

**Pull request merged**